### ISSUE TITLE
Some browsers open dataset metadata exports in a new tab as text, should open as XML/JSON

### ISSUE THREAD

**Issue by jggautier** [IQSS/dataverse#4258]
In the latest versions of Safari for Mac OS, clicking any dataset metadata exports (in Dataverse 4.9.2) opens that metadata in a new tab as text, when it should open as XML or JSON so that users can download it in the right format and so it's more readable. The JSON export opens as text in Chrome, Safari and Firefox when it should open as JSON.

Example: For https://dataverse.harvard.edu/dataset.xhtml?persistentId=doi:10.7910/DVN/2ZQQJI, clicking on Dublin Core, DDI and JSON...
<kbd>![screen shot 2017-11-03 at 8 43 23 am](https://user-images.githubusercontent.com/18374574/32374250-3a37a54e-c073-11e7-8e26-6eae2961c5e2.png)</kbd>

... opens the metadata in a new tab as text:

<kbd>![screen shot 2017-11-03 at 8 47 27 am](https://user-images.githubusercontent.com/18374574/32374358-abb1590e-c073-11e7-8e47-74956739d312.png)</kbd>

Here's a summary of what's working and isn't:

| Browser                                                  | XML (DDI and Dublin Core) | JSON (Schema.org and native Dataverse metadata) |
| -------------------------------------------------------- | ------------------------- | ----------------------------------------------- |
| Chrome (Version 68.0.3440.106 (Official Build) (64-bit)) | ✓                         | **x**                                           |
| Firefox (61.0.2 (64-bit))                                | ✓                         | **x**                                           |
| Safari (Version 11.1.2 (13605.3.8))                      | **x**                     | **x**                                           |

✓ means the browser opens the metadata export as XML or JSON
x means the browser opens the metadata export as text
Dataset page used: https://dataverse.harvard.edu/dataset.xhtml?persistentId=doi:10.7910/DVN/DE0OVP

**Comment by juancorr** [comment]
Could be the problem that there are not xml head (&lt;?xml version="1.0" encoding="UTF-8"?&gt;) )

**Comment by pameyer** [comment]
Could also be a problem with the content-type or content-disposition header.

**Comment by pameyer** [comment]
not a content-type problem: both `dcterms` and `ddi` are `application/xml` (`dataverse_json` is `text/plain` , which might be suboptimal).

**Comment by borsna** [comment]
Looks like some problem redaring the Content-Type header as @pameyer mentions.
Json should be `application/json` not `text/plain` and xml should be `application/xml` or `text/xml` (unsure witch one is the most common for xml both is mentioned in [RFC 3023](https://tools.ietf.org/html/rfc3023))

**Comment by jggautier** [comment]
Retested this since @mheppler thought it might be fixed by now (maybe as part of other dev work). It's partially fixed. The versions of Chrome and Firefox I have on my Mac recognize the JSON exports as JSON and the XML exports as XML.

The copy of Safari (13.0.2) on my laptop recognizes that the JSON exports are JSON. When I try to save the document, it adds the .json extension to the file name. But it doesn't recognize the XML exports as XML.

**Comment by jggautier** [comment]
In October 2019 I wrote that the latest version of Safari did not recognize Dataverse's XML metadata exports, like "Dublin Core" and "DataCite", as XML, which could cause problems when exporting this type of metadata through a browser.

It was the remaining issue of [the issues I reported in November 2017](https://github.com/IQSS/dataverse/issues/4258#issue-270969367) and I'm not able to reproduce it in Dataverse installations running v6.2. When I click on and save a metadata format that uses XML, it's recognized as XML:

<kbd><img width="908" alt="Screenshot 2024-05-15 at 3 55 08 PM" src="https://github.com/IQSS/dataverse/assets/18374574/1112c751-0467-432e-b30a-0d6e1882ea98"></kbd>

Closing this issue.
